### DISCUSSION TITLE
Toggle for SceneUpdate cubes to show outline only

### DISCUSSION THREAD

**Discussion by jmm-slamcore** [foxglove/community#917] [Q&A]
I'm using a message converter extension to convert a Detection3DArray message (https://github.com/ros-perception/vision_msgs/blob/ros2/vision_msgs/msg/Detection3DArray.msg) to foxglove.SceneUpdate to visualise some bounding boxes in the 3D and Image panels.

The bounding boxes are being converted to Cube Primitives which are shown as filled in cubes. Is there any way to toggle how this cube is visualized so it either shows the filled in cube or just the outline, so that I can have the filled in cube in the 3D panel and just the outline cube in the Image panel. I only see an option under the (converted) Detection3DArray topic to enable the outline (of all objects) but not disable the filled cube. I guess the CubePrimitive schema might need to change for that to work?

I think I could achieve showing just the outline by converting the bounding boxes to line primitives with the message converter as well, but then the topic will show both the cube primitive and line primitives if I understand correctly?

Any suggestions on how to achieve what I'm after? Maybe a user script that ingests the Detection3DArray topic and republishes it and then use a different message converter for each topic? One that generates cubes and one that generates lines? Or if the converter supported multiple outputs (https://github.com/orgs/foxglove/discussions/638) could use a single converter to publish different output messages?

**Comment by jtbandes** [comment]
The panel settings allow you to override the color, and you can set the alpha to zero to hide the fill completely.

<img width="307" alt="image" src="https://github.com/foxglove/community/assets/14237/d374a183-245e-4d96-8a23-a6abf93f7499">

**Reply by jmm-slamcore** [reply]
Thanks, that works for now but doesn't achieve fully what I wanted because all the outlines are black instead of the original colour of the cube (to show different classes).

**Reply by jtbandes** [reply]
Since there is no setting for the outline color, generating cubes and outlines lines separately seems like your best option right now.
